src/FS.Embedded/Manifest/ManifestParser.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace Microsoft.Extensions.FileProviders.Embedded.Manifest
+{
+    internal static class ManifestParser
+    {
+        private static readonly string DefaultManifestName = "Microsoft.Extensions.FileSystem.Embedded.Manifest.xml";
+
+        public static EmbeddedFilesManifest Parse(Assembly assembly)
+        {
+            return Parse(assembly, DefaultManifestName);
+        }
+
+        public static EmbeddedFilesManifest Parse(Assembly assembly, string name)
+        {
+            if (assembly == null)
+            {
+                throw new ArgumentNullException(nameof(assembly));
+            }
+
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            var stream = assembly.GetManifestResourceStream(name);
+            if (stream == null)
+            {
+                throw new InvalidOperationException($"Can't find a manifest with name '{name}'");
```

Comment:
Suggested wording:
```
Could not load the embedded file manifest '{manfiestName}' for assembly '{assemblyName}'.
```

Seems like this will be the error message most users hit first. You might also consider adding info to the error about how to enable manifest generation.